bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3671,21 +3671,21 @@ void ReplicaImp::onMessage<ReqMissingDataMsg>(ReqMissingDataMsg *msg) {
       }
     }
 
-    if (msg->getFullCommitIsMissing()) {
+    if (msg->getFullCommitIsMissing() && seqNumInfo.isTimeCorrect()) {
       CommitFullMsg *c = seqNumInfo.getValidCommitFullMsg();
       if (c != nullptr) {
         LOG_INFO(CNSUS, "Sending FullCommit message as a response of RFMD" << KVLOG(msgSender, msgSeqNum));
         sendAndIncrementMetric(c, msgSender, metric_sent_commitFull_msg_due_to_reqMissingData_);
       }
     }
 
-    if (msg->getFullCommitProofIsMissing() && seqNumInfo.hasFastPathFullCommitProof()) {
+    if (msg->getFullCommitProofIsMissing() && seqNumInfo.hasFastPathFullCommitProof() && seqNumInfo.isTimeCorrect()) {
```

Comment:
if time is not correct then why seqNumInfo.hasFastPathFullCommitProof() is returning true?

Reply:
We only need to avoid sending PreparePartialMsg and PartialCommitProofMsg. Have removed this check.